### PR TITLE
Print NetCDF error message if pFIO_NetCDF4_FileFormatterMod::open() fails

### PR DESCRIPTION
**Pull request by LiamBindle**
## Description

Hi all, this is a trivial update to print `nf90_strerror(status)` if `nf90_open()` in  pFIO_NetCDF4_FileFormatterMod::open() fails. 

## Motivation and Context
Currently there is not error handling if `nf90_open()` fails in pFIO_NetCDF4_FileFormatterMod::open(). As a result, a file missing read permissions will cause an error message that looks like this:
```
pe=00213 FAIL at line=00245    NetCDF4_FileFormatter.F90                <status=13>
pe=00213 FAIL at line=00090    MAPL_ExtDataCollection.F90               <status=13>
pe=00213 FAIL at line=00232    FileMetadata.F90                         <can not find time>
pe=00213 FAIL at line=00083    FileMetadataUtilities.F90                <status=1>
pe=00213 FAIL at line=02777    MAPL_ExtDataGridCompMod.F90              <status=1>
pe=00213 FAIL at line=01512    MAPL_ExtDataGridCompMod.F90              <status=1>
```
and the offending file isn't printed.

Now an error message like this is written to stderr
```
nf90_open: returned error code (13) opening ./MetDir/2016/12/MERRA2.20161214.A3dyn.05x0625.nc4 [Permission denied]
```
follow by the normal stack traces following the failed `_VERIFY(status)`.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [ ] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: pfio/NetCDF4_FileFormatter.F90 [hunk 1]
```diff
@@ -246,17 +247,18 @@ subroutine open(this, file, mode, unusable, comm, info, rc)
...
          status = nf90_open(file, IOR(omode, NF90_MPIIO), comm=this%comm, info=this%info, ncid=this%ncid)
-         !$omp end critical
-         _VERIFY(status)
       else
-         !$omp critical
          status = nf90_open(file, IOR(omode, NF90_SHARE), this%ncid)
-         !$omp end critical
-         _VERIFY(status)
       end if
+      !$omp end critical
```
Comment (mathomp4) [inline pfio/NetCDF4_FileFormatter.F90:256]: A thought from @tclune as we looked at this: Should the `!$omp end critical` come *after* the write?

**Comment by tclune** [comment]
The reason being that `write` is itself not thread-safe.  (Though STDERR might be for most compilers.)

File: pfio/NetCDF4_FileFormatter.F90 (hunk 1 above)
Comment (LiamBindle) [reply pfio/NetCDF4_FileFormatter.F90]: Thanks for the suggestion. That make sense. I fixed it.

**Comment by mathomp4** [comment]
Doing a test now with GEOS.

**Comment by mathomp4** [comment]
Well, I tested this and as far as I can see, it's zero-diff and doesn't do anything in a GEOS run.

**Review by mathomp4 — APPROVED** [review]

**Comment by mathomp4** [comment]
I've added @tclune as a reviewer as well. Just so he sees it.

**Comment by mathomp4** [comment]
@gopikrishnangs44 What version of MAPL is in GCHP 13.4.0. It's possible it has an older MAPL without the fix from @LiamBindle ?

**Comment by gopikrishnangs44** [comment]
@mathomp4  MAPL 2.6.3

**Comment by gopikrishnangs44** [comment]
I have installed the model using spack. Could you please guide me through the things necessary for a solution

**Comment by mathomp4** [comment]
Ah. Spack. Now that is harder as you don't really have the ability to hand-edit the code and recompile. Hmm. Can you provide your `logging.yaml`? We might need to add something to it. I know there is a way to get ExtData to spit out a lot of extra information. We might have to use that instead to track down the file it's having issues with.

**Comment by gopikrishnangs44** [comment]
PLease find the attached @mathomp4 
[logging.yml.txt](https://github.com/GEOS-ESM/MAPL/files/9879543/logging.yml.txt)

**Comment by mathomp4** [comment]
@gopikrishnangs44 Okay. I know how to turn on the ExtData debug prints with GEOS, I suppose you can try it with yours. In my `logging.yaml` I can change both the `console` and `CAP` handlers to `level: DEBUG`:

```yaml
   console:
      class: streamhandler
      formatter: basic
      unit: OUTPUT_UNIT
      level: DEBUG
...
   CAP:
       level: WARNING
       root_level: DEBUG
```
and get things like:
```
        EXTDATA: DEBUG: ExtData Run_: INTERP_LOOP: interpolating between bracket times, variable: AEF_ISOPRENE file: ExtData/chemistry/HEMCO/v0.0.0/sfc/MEGAN2.1_EF.geos.025x03125.esmf.nc
        EXTDATA: DEBUG:    MAPL_ExtDataInterpField: Uninterpolated field MEGAN_AEF_ISOP set to sample L 19850101 000000
        EXTDATA: DEBUG: ExtData Run_: INTERP_LOOP: interpolating between bracket times, variable: AEF_MBO file: ExtData/chemistry/HEMCO/v0.0.0/sfc/MEGAN2.1_EF.geos.025x03125.esmf.nc
        EXTDATA: DEBUG:    MAPL_ExtDataInterpField: Uninterpolated field MEGAN_AEF_MBOX set to sample L 19850101 000000
```
in my run log.

Now it looks like GCHP has extra stuff in their yaml, so maybe this won't do it? First thing to try I suppose.

NOTE: This gets *very* verbose!

**Comment by gopikrishnangs44** [comment]
I have replaced my logging with 
   console:
      class: streamhandler
      formatter: basic
      unit: OUTPUT_UNIT
      level: DEBUG
...
   CAP:
       level: WARNING
       root_level: DEBUG

and tried to run the model, but got the same error

**Comment by mathomp4** [comment]
@gopikrishnangs44 And you didn't get a zillion prints from extdata in your log?

**Comment by gopikrishnangs44** [comment]
Dear @mathomp4,
sorry for the late reply

I tried installing GCHP 14.1 with updated MAPL, v2.19. Still I am having the same issue. 

This is my jobscript for slurm:

```
#!/bin/bash
#SBATCH -J gc # name of the job
#SBATCH -p standard-low # name of the partition: available options "standard,standard-low,gpu,gpu-low,hm"
#SBATCH -n 24 # no of processes or tasks
#SBATCH -N 1 #nodes
#SBATCH --cpus-per-task=1 # no of threads per process or task
#SBATCH -t 72:00:00 # walltime in HH:MM:SS, Max value 72:00:00
#SBATCH --mem-per-cpu=8G

ulimit -c 0                  # coredumpsize
ulimit -l unlimited          # memorylocked
ulimit -u 50000              # maxproc
ulimit -v unlimited          # vmemoryuse
ulimit -s unlimited          # stacksize

source /home/21cl91p01/spack/share/spack/setup-env.sh
spack load gcc@9.3.0
spack load git@2.17.0
spack load netcdf-fortran@4.5.4
spack load cmake@3.24.2
spack load esmf@8.3.1
spack load /gmukwis          #openmpi@4.1.4
export MPI_ROOT=$(spack location -i /gmukwis)
sh /home/21cl91p01/test_14.1/setCommonRunSettings.sh
mpirun --oversubscribe -np 24 ./gchp
```


I am facing the same issue at the same time step for the new version as well.

**Comment by mathomp4** [comment]
@gopikrishnangs44 Well the answer is one of your input files is missing a date or stopped being provided or something. That's what we need to find out. 

Maybe try running with this part of your `logging.yaml`:

```yaml
   CAP.EXTDATA:
       handlers: [mpi_shared]
       level: WARNING
       root_level: DEBUG
       propagate: false
```

as well as the other debugs for CAP and CONSOLE. Perhaps this is hijacking the debug messages?

**Comment by gopikrishnangs44** [comment]
I changed logging.yml to 

`   CAP:
       level: WARNING
       root_level: DEBUG

   CAP.EXTDATA:
       handlers: [mpi_shared]
       level: WARNING
       root_level: DEBUG
       propagate: false
   console:
      class: streamhandler
#      formatter: legacy
      formatter: basic
      unit: OUTPUT_UNIT
      level: DEBUG

`


But the error is still there.

**Comment by mathomp4** [comment]
@gopikrishnangs44 At this point I'm not sure what to do as I don't have access to your run or model or machine to try things out. Some file is missing or missing data. 

You might try asking the GCHP folks on how to turn on the debug logger prints for ExtData in their model. I know how to do it in GEOS, but obviously it's different in GCHP.

Beyond that, it's adding prints to the MAPL you are building with and re-building.

**Pull request merged**